functions-python/update_feed_status/tests/conftest.py

```diff
@@ -0,0 +1,89 @@
+#
+#   MobilityData 2025
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+
+from datetime import datetime, timedelta
+from uuid import uuid4
+from shared.database_gen.sqlacodegen_models import (
+    Feed,
+    Gtfsdataset,
+)
+from test_shared.test_utils.database_utils import (
+    clean_testing_db,
+    get_testing_session,
+)
+
+future_date = datetime.now() + timedelta(days=15)
+past_date = datetime.now() - timedelta(days=15)
+
+
+def make_dataset(
+    feed_id: str, latest: bool, start: datetime, end: datetime
+) -> Gtfsdataset:
+    return Gtfsdataset(
+        id=str(uuid4()),
+        feed_id=feed_id,
+        latest=latest,
+        service_date_range_start=start,
+        service_date_range_end=end,
+    )
+
+
+def populate_database():
+    session = get_testing_session()
+
+    id_range_by_status = {
+        "inactive": (0, 6),
+        "active": (7, 10),
+        "deprecated": (11, 15),
+        "development": (16, 17),
+        "future": (18, 29),
+    }
+    for status, (a, b) in id_range_by_status.items():
+        for _id in map(str, range(a, b + 1)):
+            session.add(Feed(id=str(_id), status=status))
+
+    # -> inactive
+    for _id in [
+        "0",  # already inactive
+        "7",
+        "8",
+        "22",
+    ]:
+        session.add(make_dataset(_id, True, past_date, past_date))
+
+    # -> active
+    for _id in [
+        "2",
+        "9",  # already active
+        "12",  # deprecated
+        "16",  # development
+        "25",
+    ]:
+        session.add(make_dataset(_id, True, past_date, future_date))
+
+    # -> future
+    for _id in [
+        "10",
+    ]:
+        session.add(make_dataset(_id, False, past_date, future_date))
```

Comment:
@davidgamez @Alessandro100 Should a feed transition to "future" even if it has an active but non-latest dataset? The current logic allows it ([ref](https://github.com/MobilityData/mobility-feed-api/blob/6e8b3845a1e99610337773971f92902d7fc33dbe/functions-python/update_feed_status/src/main.py#L41)) but I don't know if it can happen.

Reply:
I'm not sure I follow, sorry. I'm basically wondering if we should set the status to "future" if a feed has two datasets: one that is current, and another (the latest) that starts in the future.

Reply:
The Feed status is based on service date range found in dataset. A feed could have many datasets, but only 1 latest dataset. The latest dataset == current (most up to date), so if the latest dataset has a service date range that starts in the future, then it will update the Feed status to future

Reply:
Got it, thank you both for clarifying!